core/src/sumeragi/mod.rs

```diff
@@ -229,11 +229,14 @@ impl SumeragiHandle {
     fn replay_block(
         block: &SignedBlock,
         wsv: &mut WorldStateView,
-        current_topology: &Topology,
+        mut current_topology: Topology,
     ) -> Topology {
-        let block = ValidBlock::validate(block.clone(), current_topology, wsv)
+        // NOTE: topology need to be updated up to block's view_change_index
+        current_topology.rotate_all_n(block.payload().header.view_change_index);
```

Comment:
Could you further explain why do we need to rotate peers here if we're doing it in `Topology::recreate_topology`?
Is it the only case where block validation would fail because of mismatched topology?

Reply:
I need this because `Topology::recreate_topology` always pass 0 as `view_change_index`. 
We need look ahead to the next block to be able get `view_change_index`.